Fix charmm patching

j3mdamas opened the pull request.

stefdoerr commented on htmd/builder/charmm.py:
```diff
@@ -228,10 +236,25 @@ def build(mol, topo=None, param=None, stream=None, prefix='structure', outdir='.
 
-    # Printing out extra patches
+    # Printing regenerable patches
+    printnoregen = 0
```
Use False and True if these are the only values you use. Don't use 1/0

stefdoerr commented on htmd/builder/charmm.py:
```diff
@@ -228,10 +236,25 @@ def build(mol, topo=None, param=None, stream=None, prefix='structure', outdir='.
+    f.write('\n')
+
+    # Printing non-regenerable patches
+    if printnoregen:
```
You could avoid this by putting the regen and no regen patches in two separate lists. I.e.
```python
for p in allpatches:
    if p.split()[1] not in noregen:
      noregenpatches.append(p)
    else:
      regenpatches.append(p)
```
and then print them separately. You avoid the flag then and seems clearer to me

stefdoerr commented:
Nice :) Made some minor comments on how to make it cleaner

On Wed, Aug 31, 2016 at 2:03 PM, João M. Damas <notifications@github.com>
wrote:

> Merged #28 <https://github.com/multiscalelab/htmd/pull/28>.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/multiscalelab/htmd/pull/28#event-773541152>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AHkVguKsQMXphxZvkdpmM4Wl_jftBaPcks5qlW2SgaJpZM4JxiG->
> .
>

j3mdamas commented:
Thanks Stefan. Corrected the first. I prefer the second like it is. For
example, to which of those list would you append the _protonationStates
list? Right now, they should all be regenerated, but the user may not need
it.

On Wed, Aug 31, 2016 at 2:38 PM, Stefan <notifications@github.com> wrote:

> Nice :) Made some minor comments on how to make it cleaner
>
> On Wed, Aug 31, 2016 at 2:03 PM, João M. Damas <notifications@github.com>
> wrote:
>
> > Merged #28 <https://github.com/multiscalelab/htmd/pull/28>.
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/multiscalelab/htmd/pull/28#event-773541152>, or mute
> > the thread
> > <https://github.com/notifications/unsubscribe-
> auth/AHkVguKsQMXphxZvkdpmM4Wl_jftBaPcks5qlW2SgaJpZM4JxiG->
> > .
>
> >
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/multiscalelab/htmd/pull/28#issuecomment-243750594>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKTzQqzF7bmVbCjlUPdjwK4ZS0mr4RWhks5qlXWugaJpZM4JxiG->
> .
>



-- 
João M. Damas
Tel:+351962414282
skype: jmdamas

stefdoerr commented:
The `_protonationStates` is already in all patches.

Even better you can do it with list comprehensions

```python
    noregenpatches = [p for p in allpatches if p.split()[1] in noregen]
    regenpatches = [p for p in allpatches if p.split()[1] not in noregen]

    # Printing regenerable patches
    if len(regenpatches) != 0:
        for p in regenpatches:
            f.write(p + '\n')
        f.write('\n')

    # Regenerate angles and dihedrals
    f.write('regenerate angles dihedrals\n')
    f.write('\n')

    # Printing non-regenerable patches
    if len(noregenpatches) != 0:
        for p in noregenpatches:
              f.write(p + '\n')
        f.write('\n')
```

Cleaner nicer cooler better :P If you don't do it I'll do it myself, hahah

j3mdamas commented:
No, that is cleaner. I can do it, thanks

On Wed, Aug 31, 2016 at 2:52 PM, Stefan <notifications@github.com> wrote:

> The _protonationStates is already in all patches.
>
> Even better you can do it with list comprehensions
>
>     noregenpatches = [p for p in allpatches if p.split()[1] in noregen]
>     regenpatches = [p for p in allpatches if p.split()[1] not in noregen]
>
>     # Printing regenerable patches
>     if len(regenpatches) != 0:
>         for p in regenpatches:
>             f.write(p + '\n')
>         f.write('\n')
>
>     # Regenerate angles and dihedrals
>     f.write('regenerate angles dihedrals\n')
>     f.write('\n')
>
>     # Printing non-regenerable patches
>     if len(noregenpatches) != 0:
>         for p in noregenpatches:
>               f.write(p + '\n')
>         f.write('\n')
>
> Cleaner nicer cooler better :P If you don't do it I'll do it myself, hahah
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/multiscalelab/htmd/pull/28#issuecomment-243753885>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKTzQviXZDU-HQQHGaXwwVTNwefUAkbDks5qlXj8gaJpZM4JxiG->
> .
>



-- 
João M. Damas
Tel:+351962414282
skype: jmdamas
